### PR TITLE
#5015: Add API GET /v2/switches/{switch-id}/flows tests

### PR DESCRIPTION
**Pull request by pkazlenka**
* Added tests for new feature (getting list of flow on switch sorted by ports)
* Added new helper method "get ports that flow uses on switch from path"()
* Added new helper method "get used ports"()

### CODE REVIEW

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesFlowsV2Spec.groovy [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    def "Empty list is returned if none of requested ports is busy with any flow"() {
+        given: "Switch with flow on it and ports this flow uses"
+        def switchUnderTest = switchPair.getDst()
+        def usedPortsList = switchHelper."get used ports"(switchUnderTest.getDpId())
+        when: "Request flows on several unused ports"
+        def unusedPortsList = new ArrayList<>(1..1000).asList() - usedPortsList
+        then: "Response is empty, but without errors"
+        switchHelper.getFlowsV2(switchUnderTest, unusedPortsList.subList(0, 3)).getFlowsByPort().isEmpty()
+    }
+
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesFlowsV2Spec.groovy:100]: Consider cases:
- 1-switch flow
- single-switch-single-port flow (will we see the same flow twice in the response?)
- down flow
- any integration with 'mirror' feature?
- should it show 'protected path' flows?

File: src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/SwitchHelper.groovy [hunk 2]
```diff
@@ -361,6 +363,15 @@ class SwitchHelper {
         return response
     }
 
+    static SwitchFlowsPerPortResponse getFlowsV2(Switch sw, List<Integer> portIds = []){
+        return northboundV2.get().getSwitchFlows(new SwitchId(sw.getDpId().getId()), portIds)
+    }
+
```
Comment (Fedichk) [inline src-java/testing/functional-tests/src/main/groovy/org/openkilda/functionaltests/helpers/SwitchHelper.groovy:369]: Please remove the redundant blank line. Thank you!

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesFlowsV2Spec.groovy [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    def "System allows to get a flow which protected path that goes through switch"() {
+        given: "Flow which protected path goes through switch"
+        when: "Get all flows going through the switch"
+        def flows = switchHelper.getFlowsV2(switchProtectedPathGoesThrough, [])
+
+        then: "The created flows (including both y-flow subflows) are in the response list from the switch"
+        flows.flowsByPort.collectMany { it.value }*.flowId
+                .contains(flowId)
+    }
+
```
Comment (Fedichk) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesFlowsV2Spec.groovy:123]: Please remove 2 redundant blank lines. Thank you!

**Review by artHalMirantis — APPROVED** [review]

File: src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesFlowsV2Spec.groovy [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    @Tidy
+    @Unroll
+    def "System allows to get a flow which protected path that goes through switch"() {
+        given: "Flow which protected path goes through switch"
+        when: "Get all flows going through the switch"
+        def flows = switchHelper.getFlowsV2(switchProtectedPathGoesThrough, [])
+
+        then: "The created flows (including both y-flow subflows) are in the response list from the switch"
```
Comment (rtretyak) [inline src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchesFlowsV2Spec.groovy:119]: Description does not correspond to the actual verification made

**Review by dmitrii-beliakov — APPROVED** [review]

**Review by rtretyak — APPROVED** [review]

**Pull request merged**